sdk/python/kfp/compiler/test_data/components/container_with_if_placeholder.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2022 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional
+
+from kfp.components import placeholders
+from kfp.dsl import container_component
+from kfp.dsl import ContainerSpec
+from kfp.dsl import OutputPath
+
+
+@container_component
+def container_with_if_placeholder(optional_input: Optional[str],
+                                  output_path: OutputPath(str)):
+    return ContainerSpec(
+        image='python:3.7',
+        command=[
+            'my_program',
+            placeholders.IfPresentPlaceholder(
+                input_name='optional_input', then='hi', else_='bye')
```

Comment:
Yes, it will still work and I will add some test cases about it too. Based on the next comment, the code should also check whether the artifact is referenced by itself additionally.